apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -734,7 +718,7 @@ wait_for_bind_or_resume({xmlstreamelement, El}, StateData) ->
                                    StateData#state{resource = R, jid = JID})
             end;
         _ ->
-            fsm_next_state(wait_for_bind_or_resume, StateData)
+            maybe_do_compress(El, wait_for_bind_or_resume, StateData)
     end;
 
 wait_for_bind_or_resume(timeout, StateData) ->
```

Comment:
I think this name is not valid any more. Now we can do `bind` or `resume` or `compression` here. How about naming it `wait_for_feature_after_auth`?